export.html

```diff
@@ -0,0 +1,226 @@
+            if (shape.node != null) {
+              const links = shape.node.getElementsByTagName('a');
+
+              for (const i = 0; i < links.length; i++)
+              {
+                const href = links[i].getAttribute('href');
+                if (href != null && graph.isCustomLink(href)) {
+                  links[i].setAttribute('href', '#');
+                }
+              }
+            }
+          };
+
+          // Fits the number of background pages to the graph
+          if (false /* PDF specific */) {
+            graph.view.getBackgroundPageBounds = function() {
+              var layout = this.graph.getPageLayout();
+              var page = this.graph.getPageSize();
+
+              return new mxRectangle(
+                  this.scale * (this.translate.x + layout.x * page.width),
+                  this.scale * (this.translate.y + layout.y * page.height),
+                  this.scale * layout.width * page.width,
+                  this.scale * layout.height * page.height);
+            };
+          }
+        }
+
+        function parseInput(input) {
+          console.debug('Parsing input')
+          const doc = mxUtils.parseXml(input);
+
+          const node = Editor.extractGraphModel(doc.documentElement, true);
+          const xmlDoc = node.ownerDocument;
+
+          if (xmlDoc.documentElement.nodeName !== 'mxfile') {
+            throw new Error(`Unsupported root node name ${xmlDoc.documentElement.nodeName}`);
+          }
+
+          return {node, xmlDoc};
+        }
+
+        function decodeDiagram(graph, xmlDoc) {
+          const codec = new mxCodec(xmlDoc);
+          const model = graph.getModel();
+          codec.decode(xmlDoc.documentElement, model);
+
+          console.debug('decoded graph doc into model', model);
+        }
+
+        function normalizeBackgroundColor(xmlDoc, format) {
+          let backgroundColor = xmlDoc.documentElement.getAttribute('background');
+
+          // Handle some different values that really mean transparent
+          if (TRANSPARENT_COLORS.includes(backgroundColor)) {
+            backgroundColor = null;
+          }
+
+          // Fall back to white for formats that don't support transparency
+          if (!TRANSPARENT_FORMATS.includes(format)) {
+            backgroundColor = '#ffffff';
+          }
+
+          return backgroundColor;
+        }
+
+        function scaleGraph(graph) {
+          const bounds = graph.getGraphBounds();
+          const pageWidth = 800
+          let scale;
+
+          if (bounds.width < pageWidth & bounds.height < 1.5 * pageWidth) {
+            scale = 4;
+          } else if (bounds.width < 2 * pageWidth & bounds.height < 3 * pageWidth) {
+            scale = 3;
+          } else if (bounds.width < 4 * pageWidth && bounds.height < 6 * pageWidth) {
+            scale = 2;
+          }
+
+          graph.view.scaleAndTranslate(scale, Math.floor(bounds.x), Math.floor(bounds.y));
```

Comment:
Can we do away with the bounds?  I'm unclear as to whether the X and Y coordinates are related to the (0, 0) of the browser viewport, the edge of the page's nearest "quadrant" (delineated by the dashed lines in the Draw.io editor) or the top-left position with content?

Reply:
Fixed this by translating to negative X and Y coordinates :see_no_evil: